src/Domain/Creators/TimeDependence/UniformRotationAboutZAxis.hpp

```diff
@@ -0,0 +1,158 @@
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/Identity.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/Rotation.hpp"
+#include "Domain/Creators/TimeDependence/GenerateCoordinateMap.hpp"
+#include "Domain/Creators/TimeDependence/TimeDependence.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace domain {
+namespace FunctionsOfTime {
+class FunctionOfTime;
+}  // namespace FunctionsOfTime
+namespace CoordinateMaps {
+namespace TimeDependent {
+template <typename Map1, typename Map2>
+class ProductOf2Maps;
+}  // namespace TimeDependent
+}  // namespace CoordinateMaps
+
+template <typename SourceFrame, typename TargetFrame, typename... Maps>
+class CoordinateMap;
+}  // namespace domain
+
+namespace Frame {
+struct Grid;
+struct Inertial;
+}  // namespace Frame
+/// \endcond
+
+namespace domain {
+namespace creators {
+namespace time_dependence {
+/*!
+ * \brief A uniform rotation about the \f$z\f$ axis:
+ * \f{eqnarray*}
+ * x &\to& x \cos \alpha(t) - y \sin \alpha(t) \\
+ * y &=\to& y \sin \alpha(t) + y \cos \alpha(t)
+ * \f},
+ * where \f$\alpha(t)\f$ is a `FunctionOfTime`. For 3 spatial dimensions,
+ * \f$z \to z\f$, and the rotation is implemented as a product of the 2D
+ * rotation and an identity map. The rotation is undefined (and therefore
+ * unimplemented here) for 1 spatial dimension.
+ */
+template <size_t MeshDim>
+class UniformRotationAboutZAxis final : public TimeDependence<MeshDim> {
+ private:
+  using Identity = domain::CoordinateMaps::Identity<1>;
+  using Rotation = domain::CoordinateMaps::TimeDependent::Rotation<2>;
+
+ public:
+  using maps_list = tmpl::list<
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, Identity>,
+      domain::CoordinateMap<Frame::Grid, Frame::Inertial, Rotation>,
+      domain::CoordinateMap<
+          Frame::Grid, Frame::Inertial,
+          CoordinateMaps::TimeDependent::ProductOf2Maps<Rotation, Identity>>>;
+
+  static constexpr size_t mesh_dim = MeshDim;
+
+  /// \brief The initial time of the function of time.
+  struct InitialTime {
+    using type = double;
+    static constexpr OptionString help = {
+        "The initial time of the function of time"};
+  };
+  /// \brief The \f$x\f$-, \f$y\f$-, and \f$z\f$-velocity.
+  struct AngularVelocity {
+    using type = double;
+    static constexpr OptionString help = {"The angular velocity of the map."};
+  };
+  /// \brief The name of the function of time to be added to the DataBox.
+  ///
+  /// The default is `"RotationAngle"`.
+  struct FunctionOfTimeName {
+    using type = std::string;
+    static constexpr OptionString help = {"Name of the function of time."};
+    static type default_value() noexcept {
+      return std::string{"RotationAngle"};
+    }
```

Comment:
Is there a strong enough reason to have this default? Generally I think it's better to have no default unless it's obvious. Else it can be difficult to understand how things are connected.

Reply:
I agree in general, but here I think it makes sense, as it does for other functions of time. Eg UniformTranslstion also sets default names. I think the logic here is that you’re unlikely to have more than one RotationAboutZAxis in your map, so standardizing the name (but allowing nonstandard names if you’re doing something unusual makes sense).

If you and @nilsdeppe agree on removing this default option, how about if I remove them across all TimeDeoendences in a separate PR?

Reply:
I definitely share your general dislike of defaults. Here it's only an issue if one is composing two `TimeDependence`s of the same type. Say you have `Translation(Rotation(Translation))`, then you need to specify the function of time names explicitly. In no other cases is there any advantage to specifying them explicitly. If we wouldn't want to support compositions I would've actually hard-coded the names to begin with. In short, I think having defaults here is fine because you almost never need to change them.